packages/js/src/ui/components/elements/OptionsDropdown/OptionsDropdown.tsx

```diff
@@ -0,0 +1,49 @@
+import { JSX, For } from 'solid-js';
+import { cn, useStyle } from '../../../helpers';
+import { DotsMenu } from '../../../icons';
+import { Button, Dropdown, dropdownTriggerButtonVariants } from '../../primitives';
+
+export type Option = { label: string; leftIcon?: JSX.Element; rightIcon?: JSX.Element };
+
+export const OptionsDropdown = (props: {
```

Comment:
In general, I would avoid this type of component here. If we need variations of dropdowns, we should be defining those where we need them. Why make a large and complicated component?

Reply:
I agree with @desiprisg. We must leverage composition at its maximum. This is the main problem we have with the current design system in the dashboard, making it very hard to use in the long run. In practise, we should follow the Radix/ShadUI pattern for the internal UI library of the Inbox component.